misc/python/materialize/parallel_workload/action.py

```diff
@@ -823,19 +823,24 @@ def run(self, exe: Executor) -> bool:
                     if len(o.columns) == 1
                     and o.columns[0].data_type == old_object.columns[0].data_type
                 ]
+            elif sink.format in ["FORMAT JSON"]:
+                # We should be able to format all data types as JSON, and they have no
+                # particular backwards-compatiblility requirements.
+                objs = [o for o in exe.db.db_objects_without_views()]
             else:
-                # multi column formats require at least as many columns as before
-                # columns also have to be of the same type, see database-issues#8385
-                objs = [
-                    o
-                    for o in exe.db.db_objects_without_views()
-                    # Webhook sources can include all headers, then we don't know the exact columns
-                    if not isinstance(old_object, WebhookSource)
-                    and not isinstance(o, WebhookSource)
-                    and len(o.columns) >= len(old_object.columns)
-                    and [c.data_type for c in o.columns[: len(old_object.columns)]]
-                    == [c.data_type for c in old_object.columns]
-                ]
+                # Avro schema migration checking can be quite strict, and we need to be not only
+                # compatible with the latest object's schema but all previous schemas.
+                # Only allow a conservative case for now: where all types and names match.
+                objs = []
```

Comment:
That's a bit unfortunate, I guess we should make the FORMAT JSON more likely

Reply:
Yeah... at least "schemas match" is a fairly common case IIUC even if it isn't the most interesting one.